packages/aws-cdk/lib/cli/cdk-toolkit.ts

```diff
@@ -264,6 +258,19 @@ export class CdkToolkit {
         const currentTemplate = templateWithNestedStacks.deployedRootTemplate;
         const nestedStacks = templateWithNestedStacks.nestedStacks;
 
+        let driftResults = undefined;
+
+        if (options.detectDrift) {
+          const env = await this.props.deployments.resolveEnvironment(stack);
+          const cfn = (await this.props.sdkProvider.forEnvironment(env, Mode.ForReading)).sdk.cloudFormation();
+
+          driftResults = await detectStackDrift(
+            cfn,
+            asIoHelper(this.ioHost, 'diff'),
+            stack.stackName,
+          );
```

Comment:
This typically goes into a helper that receives `this.props.deployments` and `this.props.sdkProvider`